src/dataObjects/dtos/singleRights/DelegationInputDto.tsx

```diff
@@ -0,0 +1,41 @@
+export enum ReduxStatusResponse {
+  Fulfilled = 'fulfilled',
+  Rejected = 'rejected',
+}
+
+export interface DelegationInputDto {
+  To: IdValuePair[];
+  Rights: DelegationRequestDto[];
+  serviceDto: ServiceDto;
+  status?: ReduxStatusResponse;
+}
+
+export class ServiceDto {
```

Comment:
To enable constructors

Reply:
Why do we need constructors though?

Reply:
It's the same reason as why we create constructors for backend code (although .Net has made it so easy to construct incomplex objects by default that in many cases it's not needed anymore). In this case though DelegationRequestDto is quite complex to construct if you're trying to do it like the second attached picture but with constructors we make it easier to construct the object. See pictures for comparing construction of objects with and without constructors, first pic is with constructors (and in first pic DelegationRequestDto is also more complex than the object we construct in the last picture). Also Redux demands that we define these objects as classes when we catch the meta-object in the fulfilled and rejected cases, and classes is harder to construct than interfaces. Redux requires it because it's a tech that can be used independently if you use Typescript or not, as classes are javascript and interfaces are Typescript.

With that being said, it doesn't make sense to only create classes or only create interfaces for future objects that we need but it needs to assesseed for each object.
Questions that can be asked when considering to create interfaces or classes with constructors:

- Do we need to catch the meta-object on the action? If yes, consider using class with constructor.
- Will there be chances of the object being created in more than one class? If yes, consider using class with constructor
- Does it make the creations of objects easier? If yes, consider using class with constructor
- Can you foresee the object being used in a non typescript use case? If yes, consider using class with constructor

![image](https://github.com/Altinn/altinn-access-management-frontend/assets/31471142/bf38135a-4503-4c43-9a23-a7dbd9610ab0)
![image](https://github.com/Altinn/altinn-access-management-frontend/assets/31471142/85a47da7-4d8f-40b7-803a-7b59734cf048)

Reply:
Great answer! I didn't know this (I've never used constructors in frontend before so this was all news to me - thanks for letting me know!) 😊